libgalois/include/katana/LC_CSR_CSC_Labeled_Graph.h

```diff
@@ -325,6 +373,73 @@ class LC_CSR_CSC_Labeled_Graph
     return std::nullopt;
   }
 
+  template <bool in_edges, enum FindEdgeWithLabelOptions opt>
+  std::optional<std::tuple<edge_iterator, edge_iterator, edge_iterator>>
+  FindFirstOrLastEdgeWithLabel(
+      GraphNode node, GraphNode key, const EdgeTy& data) const {
+    edge_iterator begin =
+        in_edges ? in_raw_begin(node, data) : raw_begin(node, data);
+    edge_iterator end = in_edges ? in_raw_end(node, data) : raw_end(node, data);
+    return FindFirstOrLastEdge<in_edges, opt>(begin, end, key);
+  }
+
+  /**
+   * Returns an edge iterator to an edge with some node and key with some label
+   * by searching for the key via the node's outgoing or incoming edges.
+   * If not found, returns nothing.
+   */
+  template <bool in_edges>
+  std::optional<edge_iterator> FindEdgeWithLabel(
+      GraphNode node, GraphNode key, const EdgeTy& data) const {
+    auto res = FindFirstOrLastEdgeWithLabel<
+        in_edges, FindEdgeWithLabelOptions::FindAny>(node, key, data);
+    if (!res) {
+      return std::nullopt;
+    }
+
+    return std::get<0>(*res);
+  }
+
+  /**
+   * Returns all edges from src to dst with some label by searching for the key
+   * via the node's outgoing or incoming edges.  If not found, returns nothing.
+   */
+  std::optional<edges_iterator> FindAllEdgesWithLabel(
+      GraphNode src, GraphNode dst, const EdgeTy& data) const {
+    // trivial check; can't be connected if degree is 0
+    if (degrees_[src] == 0 || in_degrees_[dst] == 0) {
+      return std::nullopt;
+    }
+    if (degrees_[src] < in_degrees_[dst]) {
+      return FindAllEdgesWithLabelImpl<false>(src, dst, data);
+    }
+
+    return FindAllEdgesWithLabelImpl<true>(dst, src, data);
```

Comment:
Not sure this is correct. Because these are reverse direction edges, so the semantics are different from forward direction edges.

Reply:
My understanding was that it's okay to flip the parameters when `in_edges` is true, because that's still searching in the same direction. This pattern was used in other functions.

Reply:
In other places, `bool` was returned, so that's fine. Returning an edge iterator is wrong because it is pointing to different arrays and the semantics are different.

Reply:
I see. Will fix in a follow-up PR.